CMakeLists.txt

```diff
@@ -68,57 +85,71 @@ target_include_directories(terraces
 find_package(Threads REQUIRED)
 target_link_libraries(terraces Threads::Threads)
 
-find_package(GMP)
-if(GMP_FOUND)
-	message(STATUS "GMP libraries found")
-	target_link_libraries(terraces gmpxx gmp)
-	target_compile_definitions(terraces PUBLIC USE_GMP)
-else()
-	message(STATUS "GMP libraries not found, falling back to clamped integers")
+if(TERRAPHAST_USE_GMP)
+	find_package(GMP)
+	if(GMP_FOUND)
+		message(STATUS "GMP libraries found")
+		target_link_libraries(terraces gmpxx gmp)
+		target_compile_definitions(terraces PUBLIC USE_GMP)
+	else()
+		message(STATUS "GMP libraries not found, falling back to clamped integers")
```

Comment:
Here I think it would be more appropriate to output a warning and/or fail the build

Reply:
good point.